File: natas11/solve.py
from json import dumps
from base64 import b64encode

def encrypt(data):
    key = 'qw8J'
    ordinals = []

    for index, ch in enumerate(data):
        ordinals.append(ord(ch) ^ ord(key[index % len(key)]))

    return ''.join([chr(o) for o in ordinals])

data = {
    "showpassword": "yes",
    "bgcolor": "#ffffff"
}

json = dumps(data)
encrypted_json = encrypt(json)
based_data = b64encode(bytes(encrypted_json, 'utf-8'))

print(based_data.decode('utf-8'))

File: natas11/solve.php
<?php

function xor_encrypt($in) {
    $key = 'qw8J';
    $text = $in;
    $outText = '';

    for ($i=0; $i<strlen($text); $i++) {
        $outText .= $text[$i] ^ $key[$i % strlen($key)];
    }

    return $outText;
}

$data = array(
    "showpassword" => "yes",
    "bgcolor" => "#ffffff"
);

$json_data = json_encode($data);
$encrypted_data = xor_encrypt($json_data);
$based_data = base64_encode($encrypted_data);

print($based_data);
